K2Field.K2NE.ServiceBroker/ServiceObjects/PowerShell/DynamicPowerShellSO.cs

```diff
@@ -0,0 +1,112 @@
+﻿using K2Field.K2NE.ServiceBroker.Helpers.PowerShell;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using SourceCode.SmartObjects.Services.ServiceSDK.Objects;
+using SourceCode.SmartObjects.Services.ServiceSDK.Types;
+using K2Field.K2NE.ServiceBroker.Helpers;
+using System.Data;
+
+namespace K2Field.K2NE.ServiceBroker.ServiceObjects.PowerShell
+{
+    public class DynamicPowerShellSO : ServiceObjectBase
+    {
+        public DynamicPowerShellSO(K2NEServiceBroker api)
+            : base(api)
+        {
+        }
+
+        public override string ServiceFolder
+        {
+            get
+            {
+                return Constants.ServiceFolders.PowerShell;
+            }
+        }
+
+        public string PowerShellSubdirectories
+        {
+            get
+            {
+                if (!ServiceBroker.Service.ServiceConfiguration.Contains(Constants.ConfigurationProperties.PowerShellSubdirectories))
+                {
+                    return String.Empty;
```

Comment:
I thought that if this property is empty it's not an exception. It's ok if we don't have any directories for PowerShell in ServiceBroker